packages/chrome/src/elements/header/HeaderItemIcon.tsx

```diff
@@ -5,16 +5,36 @@
  * found at http://www.apache.org/licenses/LICENSE-2.0.
  */
 
-import React, { Children, HTMLAttributes } from 'react';
+import React, {
+  Children,
+  cloneElement,
+  HTMLAttributes,
+  isValidElement,
+  PropsWithChildren,
+  ReactHTMLElement
+} from 'react';
+import { DefaultTheme, ThemeProps } from 'styled-components';
 import { StyledHeaderItemIcon } from '../../styled';
 
 /**
- * @extends HTMLAttributes<any>
+ * @extends HTMLAttributes<HTMLElement>
  */
-export const HeaderItemIcon: React.FC<HTMLAttributes<any>> = ({ children, ...props }) => {
-  // The `forwardRef` API is not needed in this element since we are cloning the provided child.
-  const Element: React.FC<unknown> = elementProps =>
-    React.cloneElement(Children.only(children as any), { ...elementProps, ...props });
+export const HeaderItemIcon = ({
+  children,
+  ...props
+}: PropsWithChildren<HTMLAttributes<HTMLElement>>) => {
+  const element = Children.only(children) as ReactHTMLElement<HTMLElement>;
 
-  return <StyledHeaderItemIcon as={Element as any} {...props} />;
+  if (isValidElement(element)) {
+    const Icon = ({
+      // eslint-disable-next-line @typescript-eslint/no-unused-vars
+      theme,
+      ...iconProps
+    }: ThemeProps<DefaultTheme> & HTMLAttributes<HTMLElement>) =>
+      cloneElement<HTMLAttributes<HTMLElement>, HTMLElement>(element, { ...props, ...iconProps });
+
+    return <StyledHeaderItemIcon as={Icon} {...props} />;
+  }
```

Comment:
Direct clone of Icon type elements is a future inconsistency that should be addressed throughout the codebase. This change is simply buttoning-up the types for props.

Reply:
Could you explain more on this matter?

Reply:
I could :wink: What are you looking for exactly? Perhaps we should talk offline. I've captured some of the inconsistencies in our version planning doc.

Reply:
Nothing in particular, was just wondering, however if we have this recorded, we should be okay.